src/new/wallet.js

```diff
@@ -765,44 +765,128 @@ class HathorWallet extends EventEmitter {
+   *   'startMiningTx': boolean to trigger start mining (default true)
    *  }
    *
    * @return {Promise} Promise that resolves when transaction is sent
    **/
-  sendManyOutputsTransaction(outputs, inputs = [], token = null, options = { changeAddress: null }) {
-    // XXX To accept here multi tokens in the same tx would be a bit more complicated because
-    // the method prepareSendTokensData would need a bigger refactor.
-    // I believe we should refactor all of that code (and it will be done on wallet-service)
-    // so I decided to change this method to accept only one token, which is enough for our needs right now
+  sendManyOutputsTransaction(outputs, inputs = [], token = null, options = {}) {
     storage.setStore(this.store);
+    const newOptions = Object.assign({ changeAddress: null, startMiningTx: true }, options);
     const txToken = token || this.token;
-    const isHathorToken = txToken.uid === HATHOR_TOKEN_CONFIG.uid;
-    const data = {
-      tokens: isHathorToken ? [] : [txToken.uid],
-      inputs: [],
-      outputs: [],
-    };
+    const tokensData = {};
+    const tokens = [];
+
+    // PrepareSendTokensData method expects all inputs/outputs for each token
+    // then the first step is to separate the inputs/outputs for each token
+    const defaultData = { outputs: [], inputs: [] };
+    if (txToken && txToken.uid !== HATHOR_TOKEN_CONFIG.uid) {
+      tokensData[txToken.uid] = Object.assign({}, defaultData);
+      tokens.push(txToken.uid);
+    } else {
+      for (const output of outputs) {
+        if (output.token && output.token !== HATHOR_TOKEN_CONFIG.uid) {
+          if (!(output.token in tokensData)) {
+            tokens.push(output.token);
+            tokensData[output.token] = Object.assign({}, defaultData);
+          }
+        }
+      }
+
+      if (tokens.length === 0) {
+        // So it's HTR
+        tokensData[HATHOR_TOKEN_CONFIG.uid] = Object.assign({}, defaultData);
+      }
+    }
 
     for (const output of outputs) {
-      data.outputs.push({address: output.address, value: output.value, tokenData: isHathorToken ? 0 : 1})
+      let tokenData;
+      if (txToken && txToken.uid !== HATHOR_TOKEN_CONFIG.uid) {
+        // A single token with third method parameter of class default
+        tokenData = 1;
+      } else {
+        if (tokens.length === 0) {
+          // HTR
+          tokenData = 0;
+        } else {
+          tokenData = tokens.indexOf(output.token) + 1;
+        }
+      }
+
+      tokensData[output.token].outputs = [...tokensData[output.token].outputs, {
+        address: output.address,
+        value: output.value,
+        timelock: output.timelock ? output.timelock : null,
+        tokenData,
+      }];
     }
 
     for (const input of inputs) {
-      data.inputs.push({tx_id: input.hash, index: input.index, token: HATHOR_TOKEN_CONFIG.uid });
+      let token;
+      if (txToken && txToken.uid !== HATHOR_TOKEN_CONFIG.uid) {
```

Comment:
Why does it override `input.token`? If it exists, shouldn't it be always used?

Reply:
Makes sense, using `input.token` when exists